airbyte-commons-temporal/src/main/java/io/airbyte/commons/temporal/NotificationUtils.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.temporal;
+
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.airbyte.commons.temporal.scheduling.ConnectionNotificationWorkflow;
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.validation.json.JsonValidationException;
+import io.temporal.client.WorkflowClient;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.util.UUID;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@NoArgsConstructor
```

Comment:
Nit:  we should avoid using this with singletons.  This is mainly because if/when we need to add annotations to the parameters, we cannot do that with the lombok generated constructor.  The suggestion here is to manually create the constructor, even if it takes no parameters at the moment for singletons.  Not a blocker to this PR.